codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesrequest.java

```diff
@@ -1767,6 +1771,12 @@ public final void setEnumType(String enumType) {
             this.enumType = enumType;
         }
 
+        @Override
+        public Builder requestOverrideConfig(AwsRequestOverrideConfig awsRequestOverrideConfig) {
+            super.requestOverrideConfig(awsRequestOverrideConfig);
+            return this;
+        }
```

Comment:
Will we ever have service-specific request override configurations? Eg. for S3's path-style-paths? We already have hooks for service-specific client configuration: https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/java/software/amazon/awssdk/services/s3/S3AdvancedConfiguration.java (just noticed, this should be renamed override, not advanced). Will that be a different method (like it is in clients)? If so, can this be in the base class instead of here?

Reply:
We don't have any now, but yeah if we ever do, then the idea is to add it as as separate property of the service base request class, e.g. `S3Request#s3RequestConfigurationOverride()`